homeassistant/components/upnp/__init__.py

```diff
@@ -126,67 +125,99 @@ async def device_discovered(
     try:
         device = await Device.async_create_device(hass, location)
     except UpnpConnectionError as err:
-        LOGGER.debug("Error connecting to device %s", location)
-        raise ConfigEntryNotReady from err
-
-    # Ensure entry has a unique_id.
-    if not entry.unique_id:
         LOGGER.debug(
-            "Setting unique_id: %s, for config_entry: %s",
-            device.unique_id,
-            entry,
+            "Error connecting to device at location: %s, err: %s", location, err
```

Comment:
It's better to pass the message to the raised ConfigEntryNotReady exception as an argument, instead of logging it. It will be logged and guarded for log spam.